src/compute-client/src/types/dataflows.rs

```diff
@@ -213,6 +213,29 @@ impl<T> DataflowDescription<OptimizedMirRelationExpr, (), T> {
         }
         panic!("GlobalId {} not found in DataflowDesc", id);
     }
+
+    /// Calls r and s on any sub-members of those types in self. Halts at the first error return.
+    pub fn walk<R, S, E>(&mut self, r: R, s: S) -> Result<(), E>
```

Comment:
Can we do some bike-shedding here? `walk` implies that the tree is traversed recursively, which this method doesn't do. It would be nice to have a name that makes that clear. Elsewhere we use `visit_children` to clarify that only direct children are visited, maybe we can use that here too?